consensus/spos/bls/subroundEndRound.go

```diff
@@ -204,6 +204,14 @@ func (sr *subroundEndRound) doEndRoundJobByLeader() bool {
 
 	// broadcast section
 
+	//TODO: Move this if before L205 before committing it, just to keep the induced hard forks system test functionality
+	if sr.Rounder().Index() > sr.RoundIndex {
+		log.Debug("doEndRoundJob: time is out -> cancel broadcasting final info and header",
```

Comment:
I do not think this will work as expected because the rounder did not manage to increment its index.

Reply:
Good catch, I will fix it

Reply:
can we add also some tolerance?
we should test with some heavy load and make sure this does not stop the producing of blocks.